prov/efa/src/rdm/rxr_pkt_pool.c

```diff
@@ -175,6 +198,118 @@ int rxr_pkt_pool_grow(struct rxr_pkt_pool *rxr_pkt_pool)
 	return 0;
 }
 
+/**
+ * @brief post multiple packet entries to receive messages from shm
+ *
+ * @param[in]		pkt_pool	pkt pool, whose usage must be SHM RX packet pool
+ * @param[in]		nrecv		number of receiving buffers to post
+ * @returns
+ * On success, return 0.
+ * On failure, return negative libfabric error code.
+ */
+static inline
+int rxr_pkt_pool_bulk_post_recv_shm(struct rxr_pkt_pool *pkt_pool, int nrecv)
+{
+	int irecv, err;
+	struct rxr_pkt_entry *pkt_entry;
+
+	assert(pkt_pool->usage == RXR_PKT_FROM_SHM_RX_POOL);
+
+	for (irecv = 0; irecv < nrecv; ++irecv) {
+		pkt_entry = rxr_pkt_entry_alloc(pkt_pool->ep, pkt_pool, RXR_PKT_FROM_SHM_RX_POOL);
+		pkt_entry->x_entry = NULL;
+
+		err = fi_recv(pkt_pool->ep->shm_ep, pkt_entry->wiredata, pkt_pool->ep->mtu_size, NULL, FI_ADDR_UNSPEC, pkt_entry);
+		if (OFI_UNLIKELY(err)) {
+			rxr_pkt_entry_release_rx(pkt_pool->ep, pkt_entry);
+			EFA_WARN(FI_LOG_EP_CTRL,
+				"failed to post buf for shm  %d (%s)\n", -err,
+				fi_strerror(-err));
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+/**
+ * @brief post multiple packet entries to receive messages from EFA device
+ *
+ * @param[in]		pkt_pool	pkt pool, whose usage must be EFA RX packet pool
+ * @param[in]		nrecv		number of receiving buffers to post
+ * @returns
+ * On success, return 0.
+ * On failure, return negative libfabric error code.
+ */
+static inline
+int rxr_pkt_pool_bulk_post_recv_efa(struct rxr_pkt_pool *pkt_pool, int nrecv)
+{
+	int irecv, err;
+	struct rxr_pkt_entry *pkt_entry;
+	struct ibv_recv_wr *head_recv_wr, *prev_recv_wr, *curt_recv_wr, *bad_wr;
```

Comment:
I think `curt_recv_wr` should be `cur_recv_wr`

Reply:
fixed in new revision.